bin/onlyswaps-verifier/src/retry_runtime.rs

```diff
@@ -0,0 +1,107 @@
+use crate::chain_state_pending::Verification;
+use alloy::primitives::FixedBytes;
+use async_stream::stream;
+use chrono::Utc;
+use config::network::NetworkConfig;
+use futures::Stream;
+use std::cmp::{Ordering, Reverse, max};
+use std::collections::{BinaryHeap, HashMap};
+use std::ops::Add;
+use std::time::Duration;
+use tokio::select;
+use tokio::sync::mpsc::{Receiver, Sender};
+
+pub struct RetryScheduler {
+    // we use a min-heap so that the soonest retry will be on top
+    to_retry: BinaryHeap<Reverse<Retry>>,
+    finality_durations: HashMap<u64, Duration>,
+    rx: Receiver<Reverse<Retry>>,
+    tx: Sender<Reverse<Retry>>,
+}
+#[derive(Debug, PartialEq, Eq)]
+pub struct Retry {
+    earliest_time: i64,
+    verification: Verification<FixedBytes<32>>,
+}
+
+impl PartialOrd for Retry {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for Retry {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.earliest_time.cmp(&other.earliest_time)
+    }
+}
+
+impl RetryScheduler {
+    pub fn new(networks: &[NetworkConfig]) -> Self {
+        let finality_durations: HashMap<u64, Duration> = networks
+            .iter()
+            .map(|it| (it.chain_id, it.finality_duration_secs))
+            .collect();
+        let (tx, rx) = tokio::sync::mpsc::channel(256);
+        Self {
+            to_retry: BinaryHeap::new(),
+            finality_durations,
+            tx,
+            rx,
+        }
+    }
+
+    pub fn tx(&self) -> Enqueuer {
+        Enqueuer {
+            tx: self.tx.clone(),
+            finality_durations: self.finality_durations.clone(),
+        }
+    }
+
+    pub fn stream(&mut self) -> impl Stream<Item = Verification<FixedBytes<32>>> {
+        stream! {
+            loop {
+                let duration_until_retry = self.to_retry.peek()
+                    .map(|it| it.0.earliest_time)
+                    .map(|secs| max(0, Utc::now().timestamp() - secs))
+                    .map(|secs| Duration::from_secs(secs as u64))
+                    .unwrap_or(Duration::MAX);
+
+                select! {
+                    task = self.rx.recv() => {
+                        if let Some(t) = task {
+                           self.to_retry.push(t);
```

Comment:
I believe you need 
```suggestion
                           self.to_retry.push(Reverse(t));
```
for your min-heap to work, no?

Reply:
`self.rx` is of type `Receiver<Reverse<Retry>>` so this isn't necessary